src/core_plugins/kibana/public/management/sections/objects/_objects.html

```diff
@@ -1,80 +1,179 @@
-                <span class="sr-only">Edit</span>
-                <i aria-hidden="true" class="fa fa-pencil"></i>
-              </button>
+    </div>
+
+    <!-- ControlledTable -->
+    <div
+      class="kuiViewContentItem kuiControlledTable"
+      ng-repeat="service in services track by $index"
+      ng-show="state.tab === service.title"
+    >
+      <!-- ToolBar -->
+      <div class="kuiToolBar">
+        <div class="kuiToolBarSearch">
+          <div class="kuiToolBarSearchBox">
+            <div class="kuiToolBarSearchBox__icon kuiIcon fa-search"></div>
+            <input
+              class="kuiToolBarSearchBox__input"
+              type="text"
+              placeholder="Search..."
+              aria-label="Filter"
+              ng-model="managementObjectsController.advancedFilter"
+            >
+          </div>
+        </div>
 
-              <button
-                ng-click="open(item)"
-                class="btn btn-info"
-                aria-label="Hide">
-                <span class="sr-only">Hide</span>
-                <i aria-hidden="true" class="fa fa-eye"></i>
-              </button>
-            </div>
+        <div class="kuiToolBarSection">
+          <!-- Bulk delete button -->
+          <button
+            class="kuiButton kuiButton--danger kuiButton--iconText"
+            confirm-click="bulkDelete()"
+            confirmation="Are you sure you want to delete the selected {{currentTab.title}}? This action is irreversible!"
+            aria-label="Delete selected objects"
+            ng-disabled="selectedItems.length == 0"
+          >
+            <span aria-hidden="true" class="kuiButton__icon kuiIcon fa-trash"></span>
+            Delete
+          </button>
 
-            <div class="pull-left">
+          <!-- Bulk export button -->
+          <button
+            class="kuiButton kuiButton--basic kuiButton--iconText"
+            ng-click="bulkExport()"
+            aria-label="Export selected objects"
+            ng-disabled="selectedItems.length == 0"
+          >
+            <span aria-hidden="true" class="kuiButton__icon kuiIcon fa-download"></span>
+            Export
+          </button>
+        </div>
+
+        <div class="kuiToolBarSection">
+          <!-- We need an empty section for the buttons to be positioned consistently. -->
+        </div>
+      </div>
+
+      <!-- NoResults -->
+      <div class="kuiPanel kuiPanel--centered" ng-if="!service.data.length">
+        <div class="kuiNoResults">
+          No {{service.title}} matched your search.
+        </div>
+      </div>
+
+      <!-- Table -->
+      <table class="kuiTable" ng-if="service.data.length">
+        <thead>
+          <tr>
+            <th class="kuiTableHeaderCell kuiTableHeaderCell--checkBox">
+              <input
+                type="checkbox"
+                class="kuiCheckBox"
+                ng-checked="currentTab.data.length > 0 && selectedItems.length == currentTab.data.length"
```

Comment:
(Possibly out-of-scope for this PR) Consider extracting this logic into a method like `allItemsChecked` on the controller so this attribute's intent becomes easier to understand.